File: Problems_and_Solutions/days-10 to 19/day-14/Java/Solution4.java
class Solution4 {
    public static int maxDepth(String s) {
        int count = 0;
        int ans = 0;
        for(char c : s.toCharArray()){
            if(c == '('){
                count++;
            }else if(c == ')'){
                count--;
            }
            ans = Math.max(ans,count);
        }
        return ans;
    }

    public static void main(String[] args) {
        String s = "(1+(2*3)+((8)/4))+1";
        System.out.println(maxDepth(s));
    }
}

File: Problems_and_Solutions/days-10 to 19/day-14/c++/solution 4.cpp
class Solution {
public:
    int maxDepth(string s) {
        int count=0;
        int max=0;
        for(char c :s)
        {
            if(c=='(')
            {
                count++;
                if(count>max)
                    max=count;
            }
            else if(c==')')
                count--;
        }
        return max;
    }
};